src/main/java/org/ow2/proactive/catalog/util/parser/WorkflowParser.java

```diff
@@ -142,38 +148,63 @@ private void addVariableIfNotNullAndModelIfNotEmpty(
         }
         if (checkIfNotNull(name, model) && checkIfNotEmpty(name, model)) {
             keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(name, model, ATTRIBUTE_VARIABLE_MODEL_LABEL));
-            addDependencyOn(keyValueMapBuilder, value, model);
+            addDependsOn(keyValueMapBuilder, value, model);
         }
     }
 
-    private void addDependencyIfCatalogObjectModelExist(
-            ImmutableList.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, TaskVariable taskVariable) {
+    private void addDependsOnIfCatalogObjectModelExistOnTaskVariable(
+            ImmutableSet.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, TaskVariable taskVariable) {
         String name = taskVariable.getName();
         String value = taskVariable.getValue();
         String model = taskVariable.getModel();
         if (checkIfNotNull(name, model) && checkIfNotEmpty(name, model)) {
-            addDependencyOn(keyValueMapBuilder, value, model);
+            addDependsOn(keyValueMapBuilder, value, model);
         }
 
     }
 
-    private void addDependencyOn(ImmutableList.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, String value,
+    private void addDependsOn(ImmutableSet.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, String value,
             String model) {
         if (model.equalsIgnoreCase(CATALOG_OBJECT_MODEL)) {
-            keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(DEPENDS_ON, value, ATTRIBUTE_DEPENDENCIES_LABEL));
+            if (model.equalsIgnoreCase(CATALOG_OBJECT_MODEL)) {
+                keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(getNameAndBucketFromDependsOn(value),
+                        getRevisionFromDependsOn(value).orElse(LATEST_VERSION),
+                        ATTRIBUTE_DEPENDS_ON_LABEL));
+            }
+        }
+    }
+
+    private String getNameAndBucketFromDependsOn(String calledWorkflow) {
+        Pattern pattern = Pattern.compile(CATALOG_OBJECT_MODEL_REGEXP);
```

Comment:
Could you please extract this line `Pattern pattern = Pattern.compile(CATALOG_OBJECT_MODEL_REGEXP);`
to final private class variable?
Because it's expensive operation and it's better not to do it every time for every variable